src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableLambdaExpression.java

```diff
@@ -0,0 +1,92 @@
+/*
+UnusedLocalVariable
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;
+
+import java.util.function.Supplier;
+
+public class InputUnusedLocalVariableLambdaExpression {
+    private final LambdaTest<String> myComponent = LambdaTest.lazy(() -> {
+        String foo = ""; // ok
+        String hoo = ""; // violation
+        new Runnable() {
+            String hoo = "";
+
+            @Override
+            public void run() {
+                String j = hoo; // violation
```

Comment:
Please define violation messages

Reply:
We do not define violation messages when the check has only one kind of violation message.

Reply:
Hmm, I thought We started in all places to make it clear that violation message meaningful on a line, and it doesn't not create extra effort on new message addition.

@rnveach , do you remember what is our rule for this now?

Reply:
We should have violation message for all violations, regardless of number of messages in check. This is a good sanity check, and will also alleviate extra work in the future if we add a new message.

Reply:
@romani @nrmancuso If it was written down somewhere (either in code or the top of issues) or controlled by CI, then it doesn't exist. If this should be a rule, then a test should enforce it.

https://github.com/checkstyle/checkstyle/issues/11214
> Single Check might have multiple violations messages, we need to see exact violation message if Check has multiple messages, to be sure that violation of correct type is placed at right line.

I consider this the main issue for this and it is documented only for checks with multiple messages. I scanned some of the checks listed and did not see any single message checks.

https://github.com/checkstyle/checkstyle/blob/2335abaea101774a23d0bdaf55f19e4f0cad0d37/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java#L118-L119
It is even documented in code that it is only for checks with multiple messages.

====================

I was not involved in the discussions or the decision to do this, so I don't know anything more than, as I said, what is written.

Reply:
Thanks a lot.
If code checks for not single message, let's keep it and Finish phase of migration to update multiple messages Checks first.
And maybe be later we will do same update for single-message, we will vote on this later on.

I recall this review item for now.

Reply:
Thanks a lot.
If code checks for not single message, let's keep it and Finish phase of migration to update multiple messages Checks first.
And maybe be later we will do same update for single-message, we will vote on this later on.

I recall this review item for now.